packages/editor/src/components/post-publish-panel/media-util.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * External dependencies
+ */
+import { v4 as uuid } from 'uuid';
+
+/**
+ * WordPress dependencies
+ */
+import { getFilename } from '@wordpress/url';
+
+// Generate a list of unique filenames given a list of URLs.
+// We want all basenames to be unique, since sometimes the extension
+// doesn't reflect the mime type, and may end up getting changed by
+// the server, on upload.
+export function generateUniqueFilenames( urls ) {
```

Comment:
As per my other comment we should not assume any extension here in this function.

We could just return a simple url=>basename object (`Record< url, basename >`) here instead of an object for each URL. Seems like an odd data structure choice

Reply:
The data structure choice makes more sense when considering the extension was being returned as well, and and array of `{ url: string, basename: string, extension: string }` is less cumbersome than `Record< string, { basename: string, extension: string } >`, particularly when iterating through the results.

I made the change since we're dropping the extension. The types and tests look cleaner now, but I'm not sure I prefer the code with all the `Object.entries()` / `Object.fromEntries()` back-and-forth 🤷